src/endpoint/virtualdevices.ts

```diff
@@ -0,0 +1,89 @@
+import { Endpoint} from '../endpoint'
+import { EndpointClient, EndpointClientConfig, HttpClientParams } from '../endpoint-client'
+import { Device, DeviceEvent, VirtualDeviceDetails } from './devices'
+import { DeviceProfileCreateRequest } from './deviceprofiles'
+import { Status, SuccessStatusValue } from '../types'
+
+
+export interface VirtualDeviceOwner {
+	ownerType: VirtualDeviceOwnerTypeEnum
+	ownerId: string
+}
+
+export enum VirtualDeviceOwnerTypeEnum {
+	USER = 'USER',
+	LOCATION = 'LOCATION'
+}
+
+export enum ExecutionTarget {
+	CLOUD = 'CLOUD',
+	LOCAL = 'LOCAL'
+}
+
+export interface VirtualDeviceCreateRequest {
+	name: string
+	owner: VirtualDeviceOwner
+	deviceProfileId?: string
+	deviceProfile?: DeviceProfileCreateRequest
+	roomId?: string
+	executionTarget?: ExecutionTarget
+	hubId?: string
+}
+
+export interface VirtualDeviceStandardCreateRequest {
+	name: string
+	owner: VirtualDeviceOwner
+	prototype: string
+	roomId?: string
+	executionTarget?: ExecutionTarget
+	hubId?: string
+}
```

Comment:
Fixed, though I don't understand why a driverId would be needed or desirable for a "prefabricated" virtual device, and how the caller would know what driver ID to include. Will clarify that, but including for now since it is in the API.

Reply:
Do you think we should leave it out for now then since we can always add it later (but removing it later would be a breaking change)?